### PR TITLE
[com_fields] Removal of unnecessary parentheses and usage of elvis operator in /administrator

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
- Remove unnecessary parentheses
- Use elvis operator!

The changes in this PR should be fairly easy to review. They are only removal of parentheses and uses of elvis instead of full ternary.

### Testing Instructions

None, should not change behavior

### Documentation Changes Required

None.

### CODE REVIEW

File: administrator/templates/isis/html/layouts/joomla/form/field/user.php [hunk 1]
```diff
@@ -118,5 +118,5 @@
 			); ?>
 		<?php endif; ?>
 	</div>
-	<input type="hidden" id="<?php echo $id; ?>_id" name="<?php echo $name; ?>" value="<?php echo (int) $value; ?>" class="field-user-input<?php ($class) ? ' ' . $class : ''; ?>" data-onchange="<?php echo $this->escape($onchange); ?>" />
+	<input type="hidden" id="<?php echo $id; ?>_id" name="<?php echo $name; ?>" value="<?php echo (int) $value; ?>" class="field-user-input<?php $class ? ' ' . $class : ''; ?>" data-onchange="<?php echo $this->escape($onchange); ?>" />
```
Comment (andrepereiradasilva) [inline administrator/templates/isis/html/layouts/joomla/form/field/user.php]: Isn't a echo needed here?

File: administrator/templates/isis/index.php [hunk 2]
```diff
@@ -247,7 +247,7 @@ function colorIsLight($color)
 		</div>
 	</header>
 <?php endif; ?>
-<?php if ((!$statusFixed) && ($this->countModules('status'))) : ?>
+<?php if ((!$statusFixed) && $this->countModules('status')) : ?>
```
Comment (andrepereiradasilva) [inline administrator/templates/isis/index.php]: More parenthisis could be removed here.

File: administrator/components/com_finder/helpers/indexer/stemmer/fr.php [hunk 3]
```diff
@@ -234,7 +234,7 @@ private static function getStem($input)
 			$rule = $vars['rules'][$rule_number];
 			preg_match($vars['rule_pattern'], $rule, $matches);
 
-			if (($matches[2] != '*') || ($intact))
+			if (($matches[2] != '*') || $intact)
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/helpers/indexer/stemmer/fr.php]: More parenthisis could be removed here

File: administrator/components/com_languages/helpers/multilangstatus.php [hunk 4]
```diff
@@ -191,12 +191,12 @@ public static function getContacts()
 				unset($warnings[$index]);
 			}
 
-			if (($warn->alang == 0) && (($warn->slang == 0) && (empty($warn->mlang))))
+			if (($warn->alang == 0) && (($warn->slang == 0) && empty($warn->mlang)))
```
Comment (andrepereiradasilva) [inline administrator/components/com_languages/helpers/multilangstatus.php]: More parenthisis could be removed here

File: administrator/components/com_languages/helpers/multilangstatus.php [hunk 5]
```diff
@@ -191,12 +191,12 @@ public static function getContacts()
...
 			{
 				unset($warnings[$index]);
 			}
 
-			if (($warn->alang == 0) && (($warn->slang == $languages) && (empty($warn->mlang))))
+			if (($warn->alang == 0) && (($warn->slang == $languages) && empty($warn->mlang)))
```
Comment (andrepereiradasilva) [inline administrator/components/com_languages/helpers/multilangstatus.php]: More parenthisis could be removed here

File: administrator/components/com_users/models/level.php [hunk 6]
```diff
@@ -57,7 +57,7 @@ protected function canDelete($record)
 				 * than the 'access' field they are on their own unfortunately.
 				 * Also make sure the table prefix matches the live db prefix (eg, it is not a "bak_" table)
 				 */
-				if ((strpos($table, $prefix) === 0) && (isset($fields['access'])))
+				if ((strpos($table, $prefix) === 0) && isset($fields['access']))
```
Comment (andrepereiradasilva) [inline administrator/components/com_users/models/level.php]: More parenthisis could be removed here

File: administrator/templates/isis/error.php [hunk 7]
```diff
@@ -192,7 +192,7 @@
 			<h1 class="page-title"><?php echo JText::_('ERROR'); ?></h1>
 		</div>
 	</header>
-	<?php if ((!$statusFixed) && ($this->getInstance()->countModules('status'))) : ?>
+	<?php if ((!$statusFixed) && $this->getInstance()->countModules('status')) : ?>
 		<!-- Begin Status Module -->
```
Comment (andrepereiradasilva) [inline administrator/templates/isis/error.php]: More parenthisis could be removed here

File: administrator/templates/isis/html/layouts/joomla/form/field/user.php (hunk 1 above)
Comment (frankmayer) [inline administrator/templates/isis/html/layouts/joomla/form/field/user.php]: Good catch!!

Comment (frankmayer) [inline administrator/templates/isis/html/layouts/joomla/form/field/user.php]: Do you want to do a PR on that?

File: administrator/components/com_languages/helpers/multilangstatus.php [hunk 8]
```diff
@@ -186,17 +186,17 @@ public static function getContacts()
...
 		foreach ($warnings as $index => $warn)
 		{
-			if (($warn->alang == 1) && ($warn->slang == 0))
+			if ($warn->alang == 1 && $warn->slang == 0)
 			{
 				unset($warnings[$index]);
 			}
 
-			if (($warn->alang == 0) && (($warn->slang == 0) && (empty($warn->mlang))))
+			if ($warn->alang == 0 && ($warn->slang == 0 && empty($warn->mlang)))
```
Comment (andrepereiradasilva) [inline administrator/components/com_languages/helpers/multilangstatus.php]: still some unneded here

File: administrator/components/com_languages/helpers/multilangstatus.php [hunk 9]
```diff
@@ -186,17 +186,17 @@ public static function getContacts()
...
 			{
 				unset($warnings[$index]);
 			}
 
-			if (($warn->alang == 0) && (($warn->slang == $languages) && (empty($warn->mlang))))
+			if ($warn->alang == 0 && ($warn->slang == $languages && empty($warn->mlang)))
```
Comment (andrepereiradasilva) [inline administrator/components/com_languages/helpers/multilangstatus.php]: still some unneded here

File: administrator/components/com_languages/helpers/multilangstatus.php (hunk 8 above)
Comment (frankmayer) [inline administrator/components/com_languages/helpers/multilangstatus.php]: Ah, yes, of course... :smile:

**Review by andrepereiradasilva — APPROVED** [review]

File: administrator/templates/isis/html/layouts/joomla/form/field/user.php (hunk 1 above)
Comment (andrepereiradasilva) [inline administrator/templates/isis/html/layouts/joomla/form/field/user.php]: ok i will do one later for this

Comment (andrepereiradasilva) [inline administrator/templates/isis/html/layouts/joomla/form/field/user.php]: done https://github.com/joomla/joomla-cms/pull/13374 please test

**Review by andrepereiradasilva — APPROVED** [review]

File: administrator/templates/isis/html/layouts/joomla/form/field/user.php (hunk 1 above)
Comment (frankmayer) [inline administrator/templates/isis/html/layouts/joomla/form/field/user.php]: It's fine. I've also installed the patch with the Joomla! Patch Tester. How do I mark your PR as tested (or code approved)?

Comment (andrepereiradasilva) [inline administrator/templates/isis/html/layouts/joomla/form/field/user.php]: go to http://issues.joomla.org/tracker/joomla-cms/13374 login with github mark your test

Comment (frankmayer) [inline administrator/templates/isis/html/layouts/joomla/form/field/user.php]: Didn't notice the Test this button, before... OK. Good to know for next time.

**Review by andrepereiradasilva — APPROVED** [review]

**Review by andrepereiradasilva — APPROVED** [review]

**Comment by frankmayer** [comment]
Conflicts resolved...

**Comment by laoneo** [comment]
Can you please solve the merge conflicts?

**Comment by frankmayer** [comment]
Conflicts resolved. Ready to be checked and merged.

File: administrator/components/com_finder/helpers/indexer/stemmer/fr.php (hunk 3 above)
Comment (Quy) [inline administrator/components/com_finder/helpers/indexer/stemmer/fr.php]: Remove parentheses `($matches[2] != '*')`/

File: administrator/templates/isis/index.php (hunk 2 above)
Comment (Quy) [inline administrator/templates/isis/index.php]: Remove parentheses `(!$statusFixed)`

File: administrator/components/com_installer/views/database/view.html.php [hunk 10]
```diff
@@ -40,8 +40,8 @@ public function display($tpl = null)
 		$this->schemaVersion = $this->get('SchemaVersion');
 		$this->updateVersion = $this->get('UpdateVersion');
 		$this->filterParams  = $this->get('DefaultTextFilters');
-		$this->schemaVersion = ($this->schemaVersion) ?  $this->schemaVersion : JText::_('JNONE');
-		$this->updateVersion = ($this->updateVersion) ?  $this->updateVersion : JText::_('JNONE');
+		$this->schemaVersion = $this->schemaVersion ?  $this->schemaVersion : JText::_('JNONE');
+		$this->updateVersion = $this->updateVersion ?  $this->updateVersion : JText::_('JNONE');
```
Comment (Quy) [inline administrator/components/com_installer/views/database/view.html.php]: Elvis operator.

File: administrator/components/com_users/models/group.php [hunk 11]
```diff
@@ -169,7 +169,7 @@ public function save($data)
 		// Check for non-super admin trying to save with super admin group
 		$iAmSuperAdmin = JFactory::getUser()->authorise('core.admin');
 
-		if ((!$iAmSuperAdmin) && ($groupSuperAdmin))
+		if ((!$iAmSuperAdmin) && $groupSuperAdmin)
```
Comment (Quy) [inline administrator/components/com_users/models/group.php]: Remove parentheses `(!$iAmSuperAdmin)`.

File: administrator/components/com_users/models/group.php [hunk 12]
```diff
@@ -193,7 +193,7 @@ public function save($data)
 
 				foreach ($otherGroups as $otherGroup)
 				{
-					$otherSuperAdmin = ($otherSuperAdmin) ? $otherSuperAdmin : JAccess::checkGroup($otherGroup, 'core.admin');
+					$otherSuperAdmin = $otherSuperAdmin ? $otherSuperAdmin : JAccess::checkGroup($otherGroup, 'core.admin');
```
Comment (Quy) [inline administrator/components/com_users/models/group.php]: Elvis operator.

File: administrator/components/com_users/models/level.php (hunk 6 above)
Comment (Quy) [inline administrator/components/com_users/models/level.php]: Remove parentheses `(strpos($table, $prefix) === 0)`

File: administrator/components/com_users/models/user.php [hunk 13]
```diff
@@ -230,7 +230,7 @@ public function save($data)
 
 			foreach ($myNewGroups as $group)
 			{
-				$stillSuperAdmin = ($stillSuperAdmin) ? ($stillSuperAdmin) : JAccess::checkGroup($group, 'core.admin');
+				$stillSuperAdmin = $stillSuperAdmin ? $stillSuperAdmin : JAccess::checkGroup($group, 'core.admin');
```
Comment (Quy) [inline administrator/components/com_users/models/user.php]: Elvis operator.

File: administrator/templates/isis/error.php (hunk 7 above)
Comment (Quy) [inline administrator/templates/isis/error.php]: Remove parentheses `(!$statusFixed)`

File: administrator/components/com_banners/tables/banner.php [hunk 14]
```diff
@@ -150,8 +150,8 @@ public function bind($array, $ignore = array())
 			$height = abs((int) $registry->get('height', 0));
 
 			// Sets the width and height to an empty string if = 0
-			$registry->set('width', ($width ? $width : ''));
-			$registry->set('height', ($height ? $height : ''));
+			$registry->set('width', ($width ?: ''));
+			$registry->set('height', ($height ?: ''));
```
Comment (Quy) [inline administrator/components/com_banners/tables/banner.php]: Remove parentheses?

File: administrator/templates/isis/index.php [hunk 15]
```diff
@@ -76,7 +76,7 @@
 $stickyToolbar = $this->params->get('stickyToolbar', '1');
 
 // Header classes
-$navbar_color = $this->params->get('templateColor') ? $this->params->get('templateColor') : '';
+$navbar_color = $this->params->get('templateColor') ?: '';
 $header_color = ($displayHeader && $this->params->get('headerColor')) ? $this->params->get('headerColor') : '';
```
Comment (Quy) [inline administrator/templates/isis/index.php]: Remove parentheses.

Comment (Quy) [inline administrator/templates/isis/index.php]: Remove parentheses.

File: administrator/templates/isis/index.php [hunk 16]
```diff
@@ -76,7 +76,7 @@
...
 $navbar_is_light = ($navbar_color && colorIsLight($navbar_color));
```
Comment (Quy) [inline administrator/templates/isis/index.php]: Remove parentheses.

File: administrator/templates/isis/index.php [hunk 17]
```diff
@@ -76,7 +76,7 @@
...
 $header_is_light = ($header_color && colorIsLight($header_color));
```
Comment (Quy) [inline administrator/templates/isis/index.php]: Remove parentheses.

File: administrator/templates/isis/login.php [hunk 18]
```diff
@@ -22,7 +22,7 @@
 $frontEndUri->setScheme(((int) $app->get('force_ssl', 0) === 2) ? 'https' : 'http');
 
 // Color Params
-$background_color = $this->params->get('loginBackgroundColor') ? $this->params->get('loginBackgroundColor') : '';
+$background_color = $this->params->get('loginBackgroundColor') ?: '';
 $color_is_light = ($background_color && colorIsLight($background_color));
```
Comment (Quy) [inline administrator/templates/isis/login.php]: Remove parentheses.

**Comment by Quy** [comment]
https://github.com/frankmayer/joomla-cms/blob/9e150680014595cbfa09d7607da36119fa644126/administrator/templates/isis/index.php#L55
Remove parentheses.

File: administrator/templates/isis/index.php (hunk 2 above)
Comment (Quy) [inline administrator/templates/isis/index.php]: @frankmayer ping

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ad35dffae7ff779ca52ac78e12585c7fe44c9342<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13370">issues.joomla.org/tracker/joomla-cms/13370</a>.</sub>

File: administrator/components/com_categories/models/fields/categoryedit.php [hunk 19]
```diff
@@ -293,7 +293,9 @@ protected function getOptions()
 					continue;
 				}
 
-				if ($option->level != 0 && isset($oldParent) && $option->value != $oldParent && !$user->authorise('core.edit.state', $assetKey))
+				if ($option->level != 0
+					&& isset($oldParent)
+					&& $option->value != $oldParent&& !$user->authorise('core.edit.state', $assetKey))
```
Comment (andrepereiradasilva) [inline administrator/components/com_categories/models/fields/categoryedit.php]: missing space `&& $option->value != $oldParent&&`
don't understand why you changed this part

File: administrator/components/com_categories/models/fields/categoryedit.php [hunk 20]
```diff
@@ -311,7 +313,9 @@ protected function getOptions()
 					continue;
 				}
 
-				if ($option->level != 0 && isset($oldParent) && $option->value != $oldParent && !$user->authorise('core.create', $assetKey))
+				if ( $option->level != 0
+					&& isset($oldParent)
+					&& $option->value != $oldParent&& !$user->authorise('core.create', $assetKey))
```
Comment (andrepereiradasilva) [inline administrator/components/com_categories/models/fields/categoryedit.php]: missing space `&& $option->value != $oldParent&&` and a space adeed after the if
don't understand why you changed this part

**Review by andrepereiradasilva — COMMENTED** [review]
code review: just two issues
the rest seems fine.

File: administrator/components/com_categories/models/fields/categoryedit.php (hunk 19 above)
Comment (frankmayer) [inline administrator/components/com_categories/models/fields/categoryedit.php]: me neither :smile_cat: (probably some autoformatting mistake)

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on f653d726f08e413e71b1bf7999a7b0ce702e1412<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13370">issues.joomla.org/tracker/joomla-cms/13370</a>.</sub>

**Comment by frankmayer** [comment]
@Quy Would you be so nice to take a look at the last minute changes and give it an OK? And thanks for all your reviews.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on f653d726f08e413e71b1bf7999a7b0ce702e1412<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13370">issues.joomla.org/tracker/joomla-cms/13370</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Pull request merged**